modules/atomic-widgets/base/atomic-widget-base.php

```diff
@@ -49,4 +55,37 @@ final public function get_stack( $with_common_controls = true ) {
 			'tabs' => [ 'content' => 'content' ],
 		];
 	}
+
+	final public function get_atomic_settings(): array {
+		$schema = static::get_props_schema();
+		$raw_settings = $this->get_settings();
+
+		$transformed_settings = [];
+
+		foreach ( $schema as $key => $prop ) {
+			if ( array_key_exists( $key, $raw_settings ) ) {
+				$transformed_settings[ $key ] = $raw_settings[ $key ];
+			} else {
+				$transformed_settings[ $key ] = $prop->get_default();
+			}
+		}
+
+		// TODO: Run through the transformers.
+		return $transformed_settings;
+	}
+
+	public static function get_props_schema(): array {
+		$cache_key = static::class;
+
+		if ( empty( static::$props_schema_cache[ $cache_key ] ) ) {
+			static::$props_schema_cache[ $cache_key ] = static::get_props_schema_definition();
+		}
+
+		return static::$props_schema_cache[ $cache_key ];
+	}
```

Comment:
Check if it really improve performance

Reply:
I'm just removing it until we get to a point we need to optimize it...